src/main/java/net/snowflake/client/jdbc/SnowflakeUtil.java

```diff
@@ -278,14 +279,23 @@ static ColumnTypeInfo getSnowflakeType(
             new ColumnTypeInfo(Types.ARRAY, defaultIfNull(extColTypeName, "ARRAY"), baseType);
         break;
 
-      case OBJECT:
       case MAP:
-        int targetType =
-            "GEOGRAPHY".equals(extColTypeName) || "GEOMETRY".equals(extColTypeName)
-                ? Types.VARCHAR
-                : Types.STRUCT;
         columnTypeInfo =
-            new ColumnTypeInfo(targetType, defaultIfNull(extColTypeName, "OBJECT"), baseType);
+            new ColumnTypeInfo(Types.STRUCT, defaultIfNull(extColTypeName, "OBJECT"), baseType);
+        break;
+
+      case OBJECT:
+        if (StructureTypeHelper.isStructureTypeEnabled()) {
+          columnTypeInfo =
+              new ColumnTypeInfo(Types.STRUCT, defaultIfNull(extColTypeName, "OBJECT"), baseType);
+        } else {
+          int targetType =
+              "GEOGRAPHY".equals(extColTypeName) || "GEOMETRY".equals(extColTypeName)
+                  ? Types.VARCHAR
+                  : StructureTypeHelper.isStructureTypeEnabled() ? Types.STRUCT : Types.VARCHAR;
```

Comment:
My suggestion is small refactor here. Could we separate the block of code for condition isStructureTypeEnabled and don,t use this flag in else block?